lib/systemd/journal/filterable.rb

```diff
@@ -88,15 +92,36 @@ def add_disjunction
       #     # current_entry is an sshd event with priority 5
       #   end
       def add_conjunction
+        @reopen_filter_conditions << :conjunction
+
         rc = Native.sd_journal_add_conjunction(@ptr)
         raise JournalError, rc if rc < 0
       end
 
       # Remove all filters and conjunctions/disjunctions.
       # @return [nil]
       def clear_filters
+        @reopen_filter_conditions = []
+
         Native.sd_journal_flush_matches(@ptr)
       end
+
+      private
+
+      def restore_filters(conditions = @reopen_filter_conditions)
```

Comment:
can we remove this default argument here? It's not used, and I think it's a little confusing at first glance.

Reply:
I'll do this after merging so it doesn't get stuck waiting for me :)